pkgs/applications/networking/irc/weechat/default.nix

```diff
@@ -117,13 +118,25 @@ in if configure == null then weechat else
         ln -s $plugin $out/plugins
       done
     '';
-  in (writeScriptBin "weechat" ''
-    #!${stdenv.shell}
-    export WEECHAT_EXTRA_LIBDIR=${pluginsDir}
-    ${lib.concatMapStringsSep "\n" (p: lib.optionalString (p ? extraEnv) p.extraEnv) plugins}
-    exec ${weechat}/bin/weechat "$@"
-  '') // {
-    name = weechat.name;
-    unwrapped = weechat;
-    meta = weechat.meta;
+
+    init = let
+      init = config.init or "";
+      scripts = builtins.concatStringsSep "\n" (map (s: "/script load ${s}") (config.scripts or []));
+    in builtins.replaceStrings [ "\n" ] [ ";" ] (scripts + init);
+
+    mkWeechat = bin: (writeScriptBin bin ''
+      #!${stdenv.shell}
+      export WEECHAT_EXTRA_LIBDIR=${pluginsDir}
+      ${lib.concatMapStringsSep "\n" (p: lib.optionalString (p ? extraEnv) p.extraEnv) plugins}
+      exec ${weechat}/bin/${bin} --run-command "${init}" "$@"
```

Comment:
If subsequent `--run-command`s overwrite the first one, maybe we could check for `--run-command` or `-r` in `$@` and concat the two. It would add a bit of complexity but just silently overwriting may confuse users.